llvm/lib/CodeGen/SelectionDAG/LegalizeVectorTypes.cpp

```diff
@@ -4809,7 +4809,8 @@ void DAGTypeLegalizer::WidenVectorResult(SDNode *N, unsigned ResNo) {
     EVT VT = N->getValueType(0);
     EVT WideVecVT = TLI.getTypeToTransformTo(*DAG.getContext(), VT);
     if (!TLI.isOperationLegalOrCustomOrPromote(N->getOpcode(), WideVecVT) &&
-        TLI.isOperationExpand(N->getOpcode(), VT.getScalarType())) {
+        (TLI.isOperationExpand(N->getOpcode(), VT.getScalarType()) ||
+         TLI.isOperationLibCall(N->getOpcode(), VT.getScalarType()))) {
```

Comment:
Add isOperationExpandOrLibCall?

Beyond the scope of this patch, but the vector legalization logic is a pain. Ideally this would be written to do vector splitting based on the available calls (which theoretically should include a vector version)